controllers/node_controller.go

```diff
@@ -26,11 +26,15 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
```

Comment:
> why? I don't understand this from the massage
The controller stored a string representation of the previous state. It skipped the reconciling in case the previous state equals to the current state (returned by `nmstatectl show`). If the nns is removed, the controller still has the previous state string. If the controller will just compare the previous state string to the new one, the nns won't be recreated.

Updated the commit message to - 
>> When squashing, this logic had to be changed since removal of an existing
>> NNS may be skipped. Now the nns is updated/re-created if the nns doesn't
>> exist or the state was changed.

I hope now it's more clear.

Reply:
it is, thanks! 

but I'm not sure you addressed the comment about section 2

Reply:
Oh, yes. Forgot to answer.
> does it mean that force-update will not update the LastSuccessfulUpdate? if so, then It's an unexpected behavior IMO.. one > would expect force-something to at least change the TS if it is the same.. no? 

That's correct, `LastSuccessfulUpdate` won't be changed in this case. I believe that's the correct behaviour. `LastSuccessfulUpdate` should represent the last time the state was updated, it the state is the same, it doesn't make sense to update it.
As I see it, the `force` is forcing  immediately checking whether the state was updated. And it is consistent with the behaviour the nns has in the other use cases, `LastSuccessfulUpdate` is updated only in case the state was actually changed.

Reply:
ah, you're see you're point. fine by me then.